api/k8s/v1/k8s.proto

```diff
@@ -288,6 +307,14 @@ message Pod {
 
   map<string, string> labels = 9;
   map<string, string> annotations = 10;
+
+  // message indicating details about why the pod is in this state.
+  // https://pkg.go.dev/k8s.io/api/core/v1#PodStatus
+  string state_reason = 11;
```

Comment:
how does this get filled? is it something you'll populate in the internal implementation?

Reply:
Yea this is a filed in the Pod Status: https://pkg.go.dev/k8s.io/api/core/v1#PodStatus